[CI] PersistentTasksExecutorIT tests failing

**Build scan**:

https://gradle-enterprise.elastic.co/s/utdg3q6rykt5e

**Repro line**:

./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.persistent.PersistentTasksExecutorIT*" -Dtests.seed=66C8283DC6B9CAFF -Dtests.locale=et-EE -Dtests.timezone=VST -Druntime.java=11

**Reproduces locally?**:

No

**Applicable branches**:

Master

**Failure history**:

https://gradle-enterprise.elastic.co/scans/failures?failures.failureClassification=all_failures&failures.failureMessage=Execution%20failed%20for%20task%20*%0A%3E%20There%20were%20failing%20tests.%20See%20the%20report%20at:%20file:///*/*/*/*/build/reports/tests/*/index.html&search.relativeStartTime=P7D&search.timeZoneId=America/Denver

**Failure excerpt**:

Examples:

```
Expected: <1> |  
-- | --
  | but: was <8> |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.junit.Assert.assertThat(Assert.java:923) |  
  | at org.elasticsearch.persistent.PersistentTasksExecutorIT.lambda$waitForTaskToStart$8(PersistentTasksExecutorIT.java:401) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:973) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:946) |  
  | at org.elasticsearch.persistent.PersistentTasksExecutorIT.waitForTaskToStart(PersistentTasksExecutorIT.java:399) |  
  | at org.elasticsearch.persistent.PersistentTasksExecutorIT.testPersistentActionWithNoAvailableNode(PersistentTasksExecutorIT.java:138) |  
  |   |  
  | java.lang.AssertionError: |  
  | Expected: <0> |  
  | but: was <8> |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.junit.Assert.assertThat(Assert.java:923) |  
  | at org.elasticsearch.persistent.PersistentTasksExecutorIT.lambda$assertNoRunningTasks$9(PersistentTasksExecutorIT.java:425) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:973) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:946) |  
  | at org.elasticsearch.persistent.PersistentTasksExecutorIT.assertNoRunningTasks(PersistentTasksExecutorIT.java:420) |  
  | at org.elasticsearch.persistent.PersistentTasksExecutorIT.cleanup(PersistentTasksExecutorIT.java:63)
```

--- a/server/src/internalClusterTest/java/org/elasticsearch/persistent/PersistentTasksExecutorIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/persistent/PersistentTasksExecutorIT.java
@@ -8,7 +8,6 @@
 
 package org.elasticsearch.persistent;
 
-import org.apache.lucene.util.LuceneTestCase;
 import org.elasticsearch.ResourceAlreadyExistsException;
 import org.elasticsearch.ResourceNotFoundException;
 import org.elasticsearch.action.support.PlainActionFuture;
@@ -34,6 +33,7 @@ import java.util.List;
 import java.util.Objects;
 
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertFutureThrows;
+import static org.hamcrest.Matchers.either;
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasSize;
@@ -42,7 +42,6 @@ import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.nullValue;
 import static org.hamcrest.core.Is.is;
 
-@LuceneTestCase.AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/75012")
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.SUITE, minNumDataNodes = 2)
 public class PersistentTasksExecutorIT extends ESIntegTestCase {
 
@@ -363,7 +362,13 @@ public class PersistentTasksExecutorIT extends ESIntegTestCase {
             // Verify that the task is STILL in internal cluster state, unassigned, with a reason indicating local abort
             PersistentTask<?> task = assertClusterStateHasTask(taskId);
             assertThat(task.getAssignment().getExecutorNode(), nullValue());
-            assertThat(task.getAssignment().getExplanation(), equalTo("Simulating local abort"));
+            // Although the assignment explanation is initially set to "Simulating local abort", because
+            // of the way we prevent reassignment to the same node in this test it may quickly change to
+            // "non cluster state condition prevents assignment" - either proves the unassignment worked
+            assertThat(task.getAssignment().getExplanation(),
+                either(equalTo("Simulating local abort"))
+                    .or(equalTo("non cluster state condition prevents assignment"))
+            );
         });
 
         // Allow it to be reassigned again

--- a/server/src/test/java/org/elasticsearch/persistent/PersistentTasksNodeServiceTests.java
+++ b/server/src/test/java/org/elasticsearch/persistent/PersistentTasksNodeServiceTests.java
@@ -318,7 +318,6 @@ public class PersistentTasksNodeServiceTests extends ESTestCase {
         assertThat(taskManager.getTasks().values(), empty());
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/75004")
     public void testTaskLocalAbort() {
         AtomicReference<String> capturedTaskId = new AtomicReference<>();
         AtomicReference<ActionListener<PersistentTask<?>>> capturedListener = new AtomicReference<>();
@@ -407,7 +406,10 @@ public class PersistentTasksNodeServiceTests extends ESTestCase {
                     equalTo("attempt to fail task [test] with id [" + persistentId + "] which has been locally aborted"));
                 break;
             case 2:
-                runningTask.markAsLocallyAborted("second local abort");
+                IllegalStateException e2 = expectThrows(IllegalStateException.class,
+                    () -> runningTask.markAsLocallyAborted("second local abort"));
+                assertThat(e2.getMessage(),
+                    equalTo("attempt to locally abort task [test] with id [" + persistentId + "] which has already been locally aborted"));
                 break;
         }
 

